app/components/questions.js

```diff
@@ -0,0 +1,108 @@
+// https://github.com/EnCiv/undebate-ssp/issues/48
+import React, { useState, useEffect } from 'react'
+import { createUseStyles } from 'react-jss'
+import CountLimitedTextInput from './count-limited-text-input'
+import Submit from './submit'
+
+export default function Questions({ electionOM, onDone }) {
+    const classes = useStyles()
+    const { electionMethods } = electionOM
+    const [questions, setQuestions] = useState([''])
+    const [error, setError] = useState('')
+    const [isValid, setIsValid] = useState(false)
+
+    const checkValid = () =>
+        !electionMethods.areQuestionsLocked() &&
+        (questions.length > 1 || (questions.length === 1 && !questions.includes('')))
+
+    useEffect(() => {
+        setIsValid(checkValid())
+    }, [isValid, electionMethods, questions])
+
+    useEffect(() => {
+        if (!questions.includes('')) {
+            setError('')
+        }
+    }, [questions, error])
+
+    const addQuestion = () => {
+        if (!questions.includes('')) {
+            setError('')
+            setQuestions([...questions, ''])
+        } else {
+            setError('All questions must not be empty!')
+        }
+    }
+
+    const generateKey = pre => {
+        return `${pre}_${new Date().getTime()}`
+    }
+
+    return (
+        <div className={classes.container}>
+            <div className={classes.send}>
+                <span>What questions would you like to ask the candidates?</span>
+                <Submit
+                    disabled={!isValid}
+                    disableOnClick
+                    onDone={() => {
+                        onDone({
+                            value: questions.filter(question => question !== ''),
+                            valid: checkValid(),
+                        })
+                    }}
+                />
+            </div>
+            <div className={classes.questionBox}>
+                {questions.map((question, index) => (
+                    <CountLimitedTextInput
+                        key={generateKey(index)}
```

Comment:
I like the idea of generate Key - but this function will generate a new key on each render, which will cause a new component to be made each time.   In this case, a good key would be the index, or what you have for the name. "Question 1".  The idea being that each time this component rerenders, that component will stay the same.  Another way to do the key would be if questions was an object, (from the above comment), then Object.entries(questions).map((key,question)=>{...}} would work here.